pkg/client/fake/client.go

```diff
@@ -523,6 +583,10 @@ func (t versionedTracker) updateObject(gvr schema.GroupVersionResource, obj runt
 }
 
 func (c *fakeClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+	if err := c.addToSchemeIfUnknownAndUnstructured(obj); err != nil {
```

Comment:
Looks like also if Partial, we should rename the func somehow

Maybe `addUnstructuredAndPartialToSchemeIfUnknown`

or 

`addToSchemeIfUnknownAndUnstructuredOrPartial`